src/EnergyPlus/WeatherManager.cc

```diff
@@ -154,59 +148,61 @@ namespace WeatherManager {
-            underwaterBoundary.OSCMIndex = UtilityRoutines::FindItemInList(underwaterBoundary.Name, state.dataSurface->OSCM);
+                                                                     ipsc->lNumericFieldBlanks,
+                                                                     ipsc->lAlphaFieldBlanks,
+                                                                     ipsc->cAlphaFieldNames,
+                                                                     ipsc->cNumericFieldNames);
+            state.dataWeather->underwaterBoundaries.emplace_back();
+            auto &underwaterBoundary = state.dataWeather->underwaterBoundaries[i - 1];
+            underwaterBoundary.Name = ipsc->cAlphaArgs(1);
+
+            ErrorObjectHeader eoh{routineName, ipsc->cCurrentModuleObject, underwaterBoundary.Name};
+            
+            underwaterBoundary.distanceFromLeadingEdge = ipsc->rNumericArgs(1);
+            underwaterBoundary.OSCMIndex = Util::FindItemInList(underwaterBoundary.Name, state.dataSurface->OSCM);
             if (underwaterBoundary.OSCMIndex <= 0) {
-                ShowSevereError(state, "Could not match underwater boundary condition object with an Other Side Conditions Model input object.");
+                ShowSevereItemNotFound(state, eoh, ipsc->cAlphaFieldNames(1), ipsc->cAlphaArgs(1));
                 errorsFound = true;
             }
-            underwaterBoundary.WaterTempScheduleIndex = ScheduleManager::GetScheduleIndex(state, state.dataIPShortCut->cAlphaArgs(2));
+            underwaterBoundary.WaterTempScheduleIndex = ScheduleManager::GetScheduleIndex(state, ipsc->cAlphaArgs(2));
             if (underwaterBoundary.WaterTempScheduleIndex == 0) {
-                ShowSevereError(
-                    state, format(R"(Water temperature schedule for "SurfaceProperty:Underwater" named "{} not found)", underwaterBoundary.Name));
+                ShowSevereItemNotFound(state, eoh, ipsc->cAlphaFieldNames(2), ipsc->cAlphaArgs(2));
                 errorsFound = true;
             }
-            if (state.dataIPShortCut->lAlphaFieldBlanks(3)) {
+
+            if (ipsc->lAlphaFieldBlanks(3)) {
                 // that's OK, we can have a blank schedule, the water will just have no free stream velocity
                 underwaterBoundary.VelocityScheduleIndex = 0;
-            } else {
-                underwaterBoundary.VelocityScheduleIndex = ScheduleManager::GetScheduleIndex(state, state.dataIPShortCut->cAlphaArgs(3));
-                if (underwaterBoundary.WaterTempScheduleIndex == 0) {
-                    ShowSevereError(
-                        state,
-                        format(R"(Free stream velocity schedule for "SurfaceProperty:Underwater" named {} not found")", underwaterBoundary.Name));
-                    errorsFound = true;
-                }
+            } else  if ((underwaterBoundary.VelocityScheduleIndex = ScheduleManager::GetScheduleIndex(state, ipsc->cAlphaArgs(3))) == 0) {
```

Comment:
Get used to seeing this idiom, i.e., assigning within an `if` condition.

Reply:
I think this is a very minor thing, but I can see two spaces between 'else' and 'if'